engine/lib/user_settings.php

```diff
@@ -125,10 +125,82 @@ function _elgg_set_user_name() {
 }
 
 /**
+ * Set a user's username
+ * Returns null if no change is required or input is not present in the form
+ * Returns true or false indicating success or failure if change was needed
+ *
+ * @return bool|void
+ *
+ * @since 3.0
+ *
+ * @access private
+ */
+function _elgg_set_user_username() {
+	$username = get_input('username');
+	$user_guid = get_input('guid');
+
+	if (!isset($username)) {
+		return;
+	}
+	
+	if (!elgg_is_admin_logged_in()) {
+		return;
+	}
+	
+	$user = get_user($user_guid);
+	if (empty($user)) {
+		return;
+	}
+	
+	if (!$user->canEdit()) {
```

Comment:
what would be the use case that an admin is logged in and he can't edit the user?